ios-base/Networking/Types/API/APIEndpoint.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+internal struct APIEndpoint: Endpoint {
+
+  private let headersProvider: HeadersProvider
+  private let endpoint: Endpoint
+
+  init(endpoint: Endpoint, headersProvider: HeadersProvider) {
+    self.endpoint = endpoint
+    self.headersProvider = headersProvider
+  }
+
+  var requestURL: URL {
+    endpoint.requestURL
+  }
+
+  var method: Network.HTTPMethod {
+    endpoint.method
+  }
+
+  var headers: [String: String] {
+    endpoint.headers + headersProvider.requestHeaders
```

Comment:
aren't endpoint.headers being override by the default headers in this case? 
I mean, shouldn't this be: 
```
headersProvider.requestHeaders + endpoint.headers

Reply:
why you say they are being overriden? The operator only joins the two dictionaries. 
If you need custom merging you could do that as well but in this case, the BaseApiClient only 'adds' extra headers to the headers provided by the endpoint. They are not supposed to conflict.

Reply:
I see what you mean I think. The custom + operator overrides the existing keys in the dictionary. This could be an issue so I will reverse the order as you suggest, but the main point is that API client headers should be related to the API/decoding/etc and the Endpoint headers related to the request.

Reply:
yup, thats what I meant 👍